### DISCUSSION TITLE
<LAB2>找不到vitis_hls輸出的IP

### DISCUSSION THREAD

**Discussion by kevin861222** [bol-edu/HLS-SOC-Discussions#82] [Q&A] [answered]
作業環境 : Ubuntu 20.04.6 LTS
問題：在 Vivado 的 block design 中找不到vitis匯出的IP

細節：
1.  已經確實在 vitis_hls 中通過各項模擬
Csim : 
![csim](https://github.com/bol-edu/HLS-SOC-Discussions/assets/79128379/4532cd32-be30-481a-adc9-6515b7e17a0c)
C syn
![csyn](https://github.com/bol-edu/HLS-SOC-Discussions/assets/79128379/be1a9c24-fb8b-4d48-9f70-952396e14637)
co-sim
![co_sim](https://github.com/bol-edu/HLS-SOC-Discussions/assets/79128379/3ff8619b-488f-429f-a97a-767493c7d814)

2. Export RTL
![finished_export_ip](https://github.com/bol-edu/HLS-SOC-Discussions/assets/79128379/5aa4fcd8-15bd-4177-a246-b0c2f7b2192b)

資料夾中也可以找的到這個檔案
![ip_in_dir](https://github.com/bol-edu/HLS-SOC-Discussions/assets/79128379/01942bc3-9961-4ca9-96e8-119d7a2d9118)

3.  在vivado 中也有導入IP
![ip_import](https://github.com/bol-edu/HLS-SOC-Discussions/assets/79128379/ed5137c3-a171-430e-8f99-b00768f90285)

4. 但是就是沒辦法找到 Fir_n11_maxi
![Screenshot from 2023-10-02 19-52-54](https://github.com/bol-edu/HLS-SOC-Discussions/assets/79128379/856c135e-aa19-4f57-a089-a3ea96db7136)

已經嘗試過關掉vivado 重新加入ip，還是找不到解決方法

為了驗證是否流程出錯，我開啟Lab1的vivado專案，發現也找不到先前導入的IP
![LAB1_ip_not_found](https://github.com/bol-edu/HLS-SOC-Discussions/assets/79128379/121a2205-2875-4ba6-abb6-20ae327bfa48)

之前用都可以順利找到IP，過程中也沒有更新ubuntu或任何軟體，請問有人知道怎麼解決這個問題嗎？
感激不盡

**Comment by LingChiYang** [comment, marked as answer]
同學你好

之前有同學也遇過這個問題，你可以檢查看看有沒有設定正確的板子型號export ip。
如果沒辦法，你可以貼出你執行的tcl。

**Reply by kevin861222** [reply]
請問是這個嗎？
![Uploading 截圖 2023-10-02 下午9.26.08.png…]()

**Reply by LingChiYang** [reply]
這邊看不到你上傳圖哦

**Reply by kevin861222** [reply]
抱歉已經重新上傳
![截圖 2023-10-02 下午9 32 30](https://github.com/bol-edu/HLS-SOC-Discussions/assets/79128379/9eb3a887-3c58-4faf-8ec2-92cb9cda00d9)

**Reply by kevin861222** [reply]
啊啊啊我想到問題出在哪裡了

**Reply by kevin861222** [reply]
我確認完更新結果～

**Reply by kevin861222** [reply]
已解決
感謝助教協助，問題原因來自於我的迷糊，不好意思，或許我需要來一點睡眠

我先前有去下載pynq-z2的board file
https://dpoauwgwqsy2x.cloudfront.net/Download/pynq-z2.zip
但我在vitis_hls選擇了xc7z020clg400-1 ，vivado選擇pynq-z2
抱歉浪費大家時間
